ls1intum/Artemis
# `Iris`: Embed chat window into exercise creation page instead of floating dialog

### ATTENTION: Only deploy to TS9!

### Checklist
#### General


#### Client

### Motivation and Context


Displaying the Iris chat window as a dialog in the "Create Programming Exercise" is not the ideal and expected way. It's better to embed it on the page.

### Description

Rewrite and refactor the chatbot widget component. In "Create Programming Exercise", the chat window will be displayed on the right side and remain visible as the page is scrolled, allowing the user to send new requests while reviewing the problem statement.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with Iris Code Editor feature enabled (you can simply use [Exercise Adaption Test Course](https://artemis-test9.artemis.cit.tum.de/course-management/10/exercises)

1. Log in to Artemis
2. Navigate to the "Create programming exercise" page (Course Management -> Course -> Exercises)
3. See that the Iris logo is there, click it
4. The chat window is on the right side of the page instead of a floating dialog box on the page
5. Resize the page, resize the chat window, and check the responsiveness
6. Scroll the page, the chat window should be visible in the same position

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots



<img width="1440" alt="Screenshot 2024-02-15 at 16 25 11" src="https://github.com/ls1intum/Artemis/assets/60965100/6aef8efe-1757-4774-bb1d-efa6fd4466f3">

<img width="1440" alt="Screenshot 2024-02-15 at 16 26 27" src="https://github.com/ls1intum/Artemis/assets/60965100/d2f99b15-a575-41c8-a364-b72aa2bb8b8f">

<img width="726" alt="Screenshot 2024-02-15 at 16 48 25" src="https://github.com/ls1intum/Artemis/assets/60965100/f2e0d00e-2dff-4ca6-9700-c64fe3773097">
